File: frontend/src/features/cup/model/event_races_cup_score.ts
import type { RaceClassResultGroupedCupScore } from '@/features/cup/model/race_class_result_grouped_cup_score'
import type { RaceOrganisationGroupedCupScore } from '@/features/cup/model/race_organisation_grouped_cup_score'
import type { SportEvent } from '@/features/event/model/sportEvent'

export interface EventRacesCupScore {
    event: SportEvent
    raceOrganisationGroupedCupScores: RaceOrganisationGroupedCupScore[]
    raceClassResultGroupedCupScores: RaceClassResultGroupedCupScore[]
}

File: backend/src/main/java/de/jobst/resulter/domain/aggregations/EventRacesCupScore.java
package de.jobst.resulter.domain.aggregations;

import de.jobst.resulter.domain.Event;
import java.util.List;

public record EventRacesCupScore(
        Event event,
        List<RaceOrganisationGroupedCupScore> raceOrganisationGroupedCupScores,
        List<RaceClassResultGroupedCupScore> raceClassResultGroupedCupScores) {}
